packages/server/src/core/degree_status/compute_bank.rs

```diff
@@ -64,51 +69,45 @@ impl<'a> DegreeStatusHandler<'a> {
                 sum_credit = bank_rule_handler.chain(chains, &mut chain_done);
                 completed = !chain_done.is_empty();
                 if completed {
-                    msg = Some(messages::completed_chain_msg(chain_done));
+                    requirement.message(messages::completed_chain_msg(chain_done));
                 }
             }
             Rule::SpecializationGroups(ref specialization_groups) => {
                 let mut groups_done_list = Vec::new();
                 sum_credit = bank_rule_handler
                     .specialization_group(specialization_groups, &mut groups_done_list);
                 completed = groups_done_list.len() >= specialization_groups.groups_number;
-                msg = Some(messages::completed_specialization_groups_msg(
-                    groups_done_list,
+                requirement.message(messages::completed_specialization_groups_msg(
+                    groups_done_list.clone(),
                     specialization_groups.groups_number,
                 ));
+                if bank.credit.is_none() {
+                    requirement
+                        .course_requirement(specialization_groups.groups_number)
+                        .course_completed(groups_done_list.len());
+                }
```

Comment:
If we switch the order of lines 84-88 with lines 80-83 we can avoid the clone in line 81.